`platform.h` error when building with GCC 13 using C++14

Working or some minor improvements, and wanted to test them using C++14. Forced it in CMakeLists.txt. This started erroring out in `platform.h` because `char8_t` is a reserved identifier in C++20. The only way to get rid of the error (which seems a false positive given the codebase) was to add

```cmake
add_compile_options(-Wno-c++20-compat)
```

CMake patch to force C++14:

```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ add_library(etl::etl ALIAS ${PROJECT_NAME})
 
 include(GNUInstallDirs)
 
+add_compile_options(-std=c++14)
+
 target_include_directories(${PROJECT_NAME} ${INCLUDE_SPECIFIER} INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
```

Error:

```
[24/303] Building CXX object test/CMakeFiles/etl_tests.dir/murmurhash3.cpp.o
FAILED: test/CMakeFiles/etl_tests.dir/murmurhash3.cpp.o 
/usr/bin/c++ -DETL_DEBUG -I/home/jaskij/projects/deps/etl/test/../include -isystem /home/jaskij/projects/deps/etl/test/UnitTest++/.. -std=gnu++20 -std=c++14 -fno-omit-frame-pointer -fno-common -Wall -Wextra -Werror -Wfloat-equal -Wuseless-cast -Wshadow -Wnull-dereference -MD -MT test/CMakeFiles/etl_tests.dir/murmurhash3.cpp.o -MF test/CMakeFiles/etl_tests.dir/murmurhash3.cpp.o.d -o test/CMakeFiles/etl_tests.dir/murmurhash3.cpp.o -c /home/jaskij/projects/deps/etl/test/murmurhash3.cpp
In file included from /home/jaskij/projects/deps/etl/test/murmurhash3.cpp:10:
/home/jaskij/projects/deps/etl/test/../include/etl/platform.h:361:25: error: identifier ‘char8_t’ is a keyword in C++20 [-Werror=c++20-compat]
  361 |   typedef uint_least8_t char8_t;
```

--- a/include/etl/platform.h
+++ b/include/etl/platform.h
@@ -358,9 +358,12 @@ SOFTWARE.
 //*************************************
 // Determine if the ETL can use char8_t type.
 #if ETL_NO_SMALL_CHAR_SUPPORT
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wc++20-compat"
   typedef uint_least8_t char8_t;
   #define ETL_HAS_CHAR8_T 1
   #define ETL_HAS_NATIVE_CHAR8_T 0
+#pragma GCC diagnostic pop
 #else
   #define ETL_HAS_CHAR8_T 1
   #define ETL_HAS_NATIVE_CHAR8_T 1

--- a/include/etl/platform.h
+++ b/include/etl/platform.h
@@ -358,12 +358,11 @@ SOFTWARE.
 //*************************************
 // Determine if the ETL can use char8_t type.
 #if ETL_NO_SMALL_CHAR_SUPPORT
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wc++20-compat"
+#include "private/diagnostic_cxx_20_compat_push.h"
   typedef uint_least8_t char8_t;
   #define ETL_HAS_CHAR8_T 1
   #define ETL_HAS_NATIVE_CHAR8_T 0
-#pragma GCC diagnostic pop
+#include "private/diagnostic_pop.h"
 #else
   #define ETL_HAS_CHAR8_T 1
   #define ETL_HAS_NATIVE_CHAR8_T 1

--- a/include/etl/private/diagnostic_cxx_20_compat_push.h
+++ b/include/etl/private/diagnostic_cxx_20_compat_push.h
@@ -1,0 +1,44 @@
+///\file
+
+/******************************************************************************
+The MIT License(MIT)
+
+Embedded Template Library.
+https://github.com/ETLCPP/etl
+https://www.etlcpp.com
+
+Copyright(c) 2023 John Wellbelove
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files(the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions :
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+******************************************************************************/
+
+/*
+ * The header include guard has been intentionally omitted.
+ * This file is intended to evaluated multiple times by design.
+ */
+
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__llvm__)
+  #pragma GCC diagnostic push 
+  #pragma GCC diagnostic ignored "-Wc++20-compat"
+#endif
+
+#if defined(__clang__) || defined(__llvm__)
+  #pragma clang diagnostic push 
+  #pragma GCC diagnostic ignored "-Wc++20-compat"
+#endif

--- a/include/etl/private/diagnostic_cxx_20_compat_push.h
+++ b/include/etl/private/diagnostic_cxx_20_compat_push.h
@@ -40,5 +40,5 @@ SOFTWARE.
 
 #if defined(__clang__) || defined(__llvm__)
   #pragma clang diagnostic push 
-  #pragma GCC diagnostic ignored "-Wc++20-compat"
+  #pragma clang diagnostic ignored "-Wc++20-compat"
 #endif

